File: language/c++/83-remove-duplicates-from-sorted-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        if(!head) return head;
        ListNode* res = head, *cur = res;
        while(head) {
            if(cur->val != head->val) {
                cur->next = head;
                cur = cur->next;
            }
            head = head->next;
        }
        cur->next = head;
        return res;
    }
};

File: language/python/83-remove-duplicates-from-sorted-list.py
# Double loop
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if head is None: 
            return head;
        res = head
        while head:
            while head.next and head.val == head.next.val:
                head.next = head.next.next
            head = head.next    
        return res    
